### PR TITLE
Fix warnings for signedness

### LINKED ISSUE #311: Fix warnings
Enable `-Wsign-conversions` and `-padantic` and fix all the warnings that the compiler will raise.

### PR DESCRIPTION
**Pull request by axic**
Part of #311.

### CODE REVIEW

File: src/eei.cpp [hunk 1]
```diff
@@ -370,20 +370,20 @@ inline int64_t maxCallGas(int64_t gas) {
 
       takeInterfaceGas(GasSchedule::base);
 
-      return Literal(static_cast<uint32_t>(m_code.size()));
+      return Literal(uint32_t(m_code.size()));
```
Comment (chfast) [inline src/eei.cpp]: Do not hide a cast.

**Comment by axic** [comment]
> Unsigned types should be avoided. They are only good for bitfields and bit manipulation.

I don't get it, why?

**Comment by axic** [comment]
Still, why?

**Comment by chfast** [comment]
I had to look it up. Most of the arguments are listed here (read all section): https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#arithmetic.

I should add that you can detect signed overflow with Undefined Behavior Sanitizer (with the newest version you can also detect unsigned overflows, but because they are valid in C++, you can get false positives). Together with fuzz testing this gives incredible results.

**Comment by axic** [comment]
As discussed the specification isn't clear what the types of these are, but implicitly it feels they are defined as unsigned in the EEI. Added an issue to review that: https://github.com/ewasm/design/issues/132

Merging this change as it doesn't changes semantics.

**Pull request merged**